### PR TITLE
simple-cipher: add exercise

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: exercises/practice/simple-cipher/.meta/example.gleam [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+) -> UtfCodepoint {
+  let shifted_codepoint_int =
+    op(
+      string.utf_codepoint_to_int(input) - min_int_code_point,
+      string.utf_codepoint_to_int(key) - min_int_code_point,
+    )
+  assert Ok(codepoint_int_shift) =
+    int.modulo(shifted_codepoint_int, string.length(alphabet))
+  assert Ok(shifted_codepoint) =
+    string.utf_codepoint(codepoint_int_shift + min_int_code_point)
```
Comment (jiegillet) [inline exercises/practice/simple-cipher/.meta/example.gleam:58]: Handling codepoints is kind of a pain isn't it? :)

File: config.json [hunk 2]
```diff
@@ -925,6 +925,22 @@
         ],
         "difficulty": 4
       },
+      {
+        "slug": "simple-cipher",
+        "name": "Simple Cipher",
+        "uuid": "37dd415b-1d67-434d-8705-19295d1438b8",
+        "practices": [
+          "random",
+          "iterator",
```
Comment (jiegillet) [inline config.json:934]: Let's add these in prerequisistes too.

**Review by jiegillet — COMMENTED** [review]
Very nice!
I left some small comments

File: exercises/practice/simple-cipher/.meta/example.gleam [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+pub fn decode(ciphertext ciphertext: String, key key: String) -> String {
+  shift(input: ciphertext, key: key, by: fn(x, y) { x - y })
+}
+
+pub fn generate_key() -> String {
+  let alphabet_codepoints = string.to_utf_codepoints(alphabet)
+
+  iterator.repeatedly(fn() { int.random(0, list.length(alphabet_codepoints)) })
```
[the remark this answers is not in the document]
Comment (lpil) [reply exercises/practice/simple-cipher/.meta/example.gleam:18]: It would, aye!

[the remark this answers is not in the document]
Comment (jiegillet) [reply exercises/practice/simple-cipher/.meta/example.gleam:18]: Oh, TIL, I always thought `string.length` was O(1).

File: exercises/practice/simple-cipher/.meta/example.gleam (hunk 1 above)
Comment (ErikSchierboom) [reply exercises/practice/simple-cipher/.meta/example.gleam:58]: Yep! Somewhat annoying :)

File: exercises/practice/simple-cipher/test/simple_cipher_test.gleam [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+pub fn substitution_cipher_can_decode_messages_longer_than_the_key_test() {
+  simple_cipher.decode(ciphertext: "iboaqcnecbfcr", key: "abc")
+  |> should.equal("iamapandabear")
+}
+
+pub fn random_key_cipher_key_is_made_only_of_lowercase_letters_test() {
+  assert Ok(re) = regex.from_string("^[a-z]+$")
+  assert True =
+    simple_cipher.generate_key()
+    |> regex.check(re, _)
```
[the remark this answers is not in the document]
Comment (ErikSchierboom) [reply exercises/practice/simple-cipher/test/simple_cipher_test.gleam:56]: We should!

File: exercises/practice/simple-cipher/.meta/example.gleam [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  |> iterator.map(fn(shift) { string.utf_codepoint(min_int_code_point + shift) })
+  |> iterator.take(up_to: 100)
+  |> iterator.to_list()
```
[the remark this answers is not in the document]
Comment (ErikSchierboom) [reply exercises/practice/simple-cipher/.meta/example.gleam:21]: Yep. I didn't see another way to easy do this without creating a list of a 100 elements first. But I'm open to suggestions!

File: exercises/practice/simple-cipher/.meta/example.gleam [hunk 6]
```diff
@@ -0,0 +1,65 @@
+import gleam/int
+import gleam/list
+import gleam/string
+import gleam/result
+import gleam/iterator
+
+pub fn encode(plaintext plaintext: String, key key: String) -> String {
+  shift(input: plaintext, key: key, by: fn(x, y) { x + y })
```
[the remark this answers is not in the document]
Comment (ErikSchierboom) [reply exercises/practice/simple-cipher/.meta/example.gleam:8]: Nice

File: exercises/practice/simple-cipher/.meta/example.gleam (hunk 3 above)
[the remark this answers is not in the document]
Comment (ErikSchierboom) [reply exercises/practice/simple-cipher/.meta/example.gleam]: As the alphabet length is fairly constant, I've made it a, eh, constant :)

[the remark this answers is not in the document]
Comment (lpil) [reply exercises/practice/simple-cipher/.meta/example.gleam]: Unfortunately the size of a character isn't fixed so we can't tell how many characters there are from the byte size of a string. We are lacking a function to get the byte size though, perhaps would be a good stdlib addition. Could be confusing though as JS is utf-16 and Erlang is utf-8.

**Review by jiegillet — APPROVED** [review]
Already approving, but there is one thing left to fix

**Pull request merged**